src/js/utils/hooks.ts

```diff
@@ -91,3 +94,39 @@ export function useUrlSearchParams(key: string, defaultValue?: string): [string,
 
     return [value, (value: string) => updateUrlSearchParams(value, key, history)];
 }
+
+/**
+ * Hook for validating selected files from paginated data
+ *
+ * @param data - The data files
+ * @param fetchNextPage - Fetches the next page of data
+ * @param hasNextPage - Whether the search result has a next page
+ * @param isLoading - Whether the data is loading
+ * @param onClick - A callback function to handle file selection
+ * @param selected - The selected file id
+ */
+export function useValidateFiles(
+    data: InfiniteData<FileResponse>,
+    fetchNextPage: (options?: FetchNextPageOptions) => Promise<InfiniteQueryObserverResult>,
+    hasNextPage: boolean,
+    isLoading: boolean,
+    onClick: (file: string) => void,
+    selected: string,
+) {
```

Comment:
I would redesign the hook around it fetching the list of files for itself. Doing it this way ensures that you get a full list of files even if the user has applied a filter and if there is no filter applied then react-query's caching should mean that the request results are re-used between both component automagically.

signature would look along the lines of:

```suggestion
export function useValidateFiles(
    type: string
    selected: string[]
    setSelected: (selected: string[] => void)
) {
```